apps/web/controllers/users/registration.rb

```diff
@@ -7,8 +6,19 @@ class Registration
     include Web::Controllers::Authentication::Skip
 
     params do
+      configure do
+        def valid_format?(email)
+          email =~ /@/
+        end
+
+        # TODO: extract an interactor
+        def unique?(email)
+          UserRepository.new.by_email(email).nil?
```

Comment:
Next step:
- Create an interactor for checking email uniqueness and use it instead of calling UserRepo instance method
- Inject the interactor as dependency